frontend/src/components/homepage/Filter.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+
+import ToggleButton from "../navigation/ToggleButton";
+import "./Filter.css";
+
+export type FilterProps = {
+  approvedLanguages: { [name: string]: number };
+  level: number;
+  setLevel: (newState: number) => void;
+  language: string;
+  setLanguage: (newState: string) => void;
+  role: boolean;
+  setRole: (newState: boolean) => void;
+};
+
+const Filter = ({
+  approvedLanguages,
+  level,
+  setLevel,
+  language,
+  setLanguage,
+  role,
+  setRole,
+}: FilterProps) => {
+  // language state in Homepage can't default to user context value
+  // set the default language from user approvedLanguages here
+  if (!Object.keys(approvedLanguages).includes(language)) {
+    const firstLang = Object.keys(approvedLanguages)[0];
+    setLanguage(firstLang);
+    setLevel(approvedLanguages[firstLang]);
+  }
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setLanguage(event.target.value);
+  };
+  const handleRangeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setLevel(parseInt(event.target.value, 10));
+  };
+  const languageOptions = Object.keys(approvedLanguages).map((lang) => (
+    <option value={lang}>{lang}</option>
+  ));
+  const maxLvl = approvedLanguages[language];
+  const lvlOptions = [];
+  for (let i = 1; i <= maxLvl; i += 1) {
+    lvlOptions.push(<option value={i.toString()} label={i.toString()} />);
+  }
+  return (
+    <div className="filter">
+      <h1>Filter Stories</h1>
+      <div>
+        <h4>TRANSLATION LANGUAGE</h4>
+        <select
+          name="language"
+          id="language"
+          value={language}
+          onChange={handleSelectChange}
+        >
+          {languageOptions}
+        </select>
+      </div>
+      <div>
+        <h4>ROLE</h4>
+        <ToggleButton
+          leftStateIsSelected={role}
+          leftStateLabel="For Translation"
+          rightStateLabel="For Review"
+          onToggle={setRole}
+        />
+      </div>
+      <div id="filter-level">
```

Comment:
my level slider doesn't seem to align properly with the levels--is this an issue with my laptop?
![image](https://user-images.githubusercontent.com/32009013/123519593-4eb87e00-d67a-11eb-8fe0-e27a666d495a.png)

also quick q i think i'm having some trouble navigating the figma hahah is there an updated wireframe where the levels are a slider instead of pill buttons?

Reply:
Made a change, hows it look now?

slider doesnt have a figma design right now - Russel and I discussed using sliders over buttons on slack. Currently the slider's styling is default html/css.

Reply:
still doesn't seem to match up properly for me
![image](https://user-images.githubusercontent.com/32009013/123529761-ab8b5700-d6c1-11eb-9b23-39103f3531d0.png)

i'm not sure if my laptop's aspect ratio is just weird though?